client-js/app/elements/labrad-plot1d.ts

```diff
@@ -252,4 +258,46 @@ export class Plot1D extends polymer.Base implements Plot {
          .attr('d', this.line);
     }
   }
+  
+  private reset() {
+    this.xScale.domain([this.limits.xMin, this.limits.xMax]);
+    this.yScale.domain([this.limits.yMin, this.limits.yMax]);
+    this.xAxis.scale(this.xScale);
+    this.yAxis.scale(this.yScale);
+    this.zoom.x(this.xScale);
+    this.zoom.y(this.yScale);
+    this.handleZoom();
+  }
+
+  private zoomRectangle() {
+    var e = this,
+        origin = d3.mouse(e),
```

Comment:
I would suggest using destructuring assignment to pull out the x and y components, instead of using indices throughout:
```javascript
[originX, originY] = d3.mouse(e),
```